File: src/codes/single-state-handling/ts/uncode.ts
class Document {
  private state: string = "draft"; // 状態: 'draft', 'review', 'published'

  publish() {
    if (this.state === "draft") {
      console.log("レビュー依頼を送信");
      this.state = "review";
    } else if (this.state === "review") {
      console.log("公開しました");
      this.state = "published";
    } else if (this.state === "published") {
      console.log("すでに公開済みです");
    }
  }
}

File: src/codes/single-state-handling/py/uncode.py
class Document:
    def __init__(self):
        self.state = "draft"  # 状態: 'draft', 'review', 'published'

    def publish(self):
        if self.state == "draft":
            print("レビュー依頼を送信")
            self.state = "review"
        elif self.state == "review":
            print("公開しました")
            self.state = "published"
        elif self.state == "published":
            print("すでに公開済みです")
